### PR TITLE
Infrastructure Permissions

### PR DESCRIPTION
**Pull request by srivastava-jyoti**
Targets #1162

### CODE REVIEW

File: Modules/Infrastructure/Resources/views/index.blade.php [hunk 1]
```diff
@@ -2,21 +2,16 @@
...
+    @if(auth()->user()->can('infrastructure.backups.view'))
+        @include('infrastructure::menu_header', ['tab' => 'DataBase Backups'])
+
+        <ul class="mt-5 w-50p list list-group">
+            @foreach($storageBuckets as $storageBucket)
+                <li class="list-item list-group-item">{{ \Str::title($storageBucket['Name']) }}</li>
+            @endforeach
+        </ul>
+    @else
+        @include('errors.403')
```
Comment (GauravGusain98) [inline Modules/Infrastructure/Resources/views/index.blade.php:14]: This permission should not be handled inside the blade. You should check the permission on the backend.

We don't want the user to reach the blade file if he doesn't have permission.

File: Modules/Infrastructure/Resources/views/instances.blade.php [hunk 2]
```diff
@@ -2,12 +2,16 @@
...
-        <li class="list-item list-group-item">{{ \Str::title($instance['Instances'][0]['Tags'][0]['Value']) }}</li>
-        @endforeach
-    </ul>
+        <ul class="mt-5 w-50p list list-group">
+            @foreach($instances as $instance)
+            <li class="list-item list-group-item">{{ \Str::title($instance['Instances'][0]['Tags'][0]['Value']) }}</li>
+            @endforeach
+        </ul>
+    @else
+        @include('errors.403')
```
Comment (GauravGusain98) [inline Modules/Infrastructure/Resources/views/instances.blade.php:14]: Same here

File: Modules/Infrastructure/Resources/views/index.blade.php (hunk 1 above)
Comment (GauravGusain98) [reply Modules/Infrastructure/Resources/views/index.blade.php:14]: For this, we usually use `Policies` in Laravel. You can check it in the documentation.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
minor feedbacks

File: app/Providers/AuthServiceProvider.php [hunk 3]
```diff
@@ -13,7 +13,6 @@ class AuthServiceProvider extends ServiceProvider
      * @var array
      */
     protected $policies = [
-        'App\Models\Client' => 'App\Policies\ClientPolicy',
```
[the remark this answers is not in the document]
Comment (srivastava-jyoti) [reply app/Providers/AuthServiceProvider.php]: It was done by mistake.

File: Modules/Infrastructure/Resources/views/menu_header.blade.php [hunk 4]
```diff
@@ -3,14 +3,12 @@
         <li class="nav-item">
             <a class="nav-item nav-link font-weight-bold ml-0 pl-2 {{ $tab == 'DataBase Backups' ? 'active' : ''}}" href="{{ route('infrastructure.index') }}"><i class="fa fa-users"></i>&nbsp;DataBase Backups</a>
         </li>
-
         <li class="nav-item">
             <a class="nav-item nav-link font-weight-bold {{ $tab == 'EC2 Instances' ? 'active' : ''}}" href="{{ route('infrastructure.get-instances') }}"><i class="fa fa-users"></i>&nbsp;EC2 Instances</a>
         </li>
-
-    </ul>
+     </ul>
```
Comment (GauravGusain98) [inline Modules/Infrastructure/Resources/views/menu_header.blade.php:9]: This tag is not aligned properly.

Comment (GauravGusain98) [inline Modules/Infrastructure/Resources/views/menu_header.blade.php:9]: This tag is not aligned properly.

File: Modules/Infrastructure/Http/Controllers/InfrastructureController.php [hunk 5]
```diff
@@ -19,20 +21,24 @@ public function __construct(InfrastructureServiceContract $service)
...
+        $this->authorize('Ec2Instancesview', $this);
         $instances = $this->service->getServersInstances();
 
         return view('infrastructure::instances')->with('instances', $instances);
     }
 
     public function getBillingDetails()
     {
+        $this->authorize('Billingview', $this);
+
```
Comment (GauravGusain98) [inline Modules/Infrastructure/Http/Controllers/InfrastructureController.php:41]: remove this extra line

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Minor feedbacks, else looks good.

File: Modules/Infrastructure/Http/Controllers/InfrastructureController.php (hunk 5 above)
Comment (srivastava-jyoti) [reply Modules/Infrastructure/Http/Controllers/InfrastructureController.php:41]: PHP cs check is not getting passed on removing the extra line.

**Review by GauravGusain98 — APPROVED** [review]
looks fine

**Pull request merged**